cypress/plugins/pluginUtils.js

```diff
@@ -115,4 +115,37 @@ function generateIndexFile(path, outputObj) {
   }
 }
 
-module.exports = { getAndDereferenceOpenRpc, generateIndexFile };
+/**
+ * @function preprocessDeviceData
+ * @description Reads the device data JSON file and adds it to the config object.
+ * @param {string} config - The config object.
+ * @example
+ * preprocessDeviceData(config);
+ */
+
+function preprocessDeviceData(config) {
+  const deviceMac = config.env.deviceMac;
+  try {
+    if (!deviceMac) {
+      throw new Error('Device MAC address is required.');
+    }
+    const formattedDeviceMac = deviceMac.replace(/:/g, '').toUpperCase();
+    const jsonFilePath = `cypress/fixtures/external/devices/${formattedDeviceMac}.json`;
+    let deviceData;
+
+    try {
+      deviceData = JSON.parse(fs.readFileSync(jsonFilePath, 'utf-8'));
+    } catch (readError) {
+      throw new Error(
+        `Error reading or parsing the JSON file at ${jsonFilePath}: ${readError.message}`
```

Comment:
Same here, I don't think we should not run if a device file doesn't exist.

Reply:
Replaced with logger.error